airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -13,6 +16,29 @@ const getDefaultCursorField = (streamNode: SyncSchemaStream): string[] => {
   return streamNode.config?.cursorField || [];
 };
 
+const cleanBreakingFieldChanges = (streamNode: SyncSchemaStream, breakingChangesByStream: StreamTransform[]) => {
+  breakingChangesByStream.forEach((streamTransformation) => {
+    // get all of the removed field paths
+    const removedFieldPaths = streamTransformation.updateStream?.map((update) => update.fieldName);
+
+    if (streamNode.config?.primaryKey && removedFieldPaths?.length) {
+      // if any of the field paths in the primary key match any of the field paths that were removed, clear the entire primaryKey property
+      if (
+        streamNode.config.primaryKey.some((key) => removedFieldPaths.some((removedPath) => isEqual(key, removedPath)))
```

Comment:
it's possible that [lodash intersection](https://lodash.com/docs/#intersection) could tidy this up, but i haven't figured out how